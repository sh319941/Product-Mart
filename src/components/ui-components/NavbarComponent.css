   .hidden.menu {
      display: none;
    }

    .masthead.segment {
      min-height: 700px;
      padding: 1em 0;
    }
    .masthead .logo.item img {
      margin-right: 1em;
    }
    .masthead .ui.menu .ui.button {
      margin-left: 0.5em;
    }
 

    .ui.vertical.stripe {
      padding: 8em 0;
    }
    .ui.vertical.stripe .floated.image {
      clear: both;
    }
    .ui.vertical.stripe p {
      font-size: 1.33em;
    }
    .ui.vertical.stripe .horizontal.divider {
      margin: 3em 0;
    }

    .quote.stripe.segment {
      padding: 0;
    }
    .quote.stripe.segment .grid .column {
      padding-top: 5em;
      padding-bottom: 5em;
    }

    .footer.segment {
      padding: 5em 0;
    }

    .secondary.pointing.menu .toc.item {
      display: none;
    }

    @media only screen and (max-width: 700px) {
      .ui.fixed.menu {
        display: none !important;
      }
      .secondary.pointing.menu .item,
      .secondary.pointing.menu .menu {
        display: none;
      }
      .secondary.pointing.menu .toc.item {
        display: block;
      }
      .masthead.segment {
        min-height: 350px;
      }
    }
    .link, ddm {
    color:steelblue;
    font-weight: bold;
    font-size: 1.2rem;
  }
  .link:hover {
    text-decoration: none;
    font-weight: bolder;
    background: rgba(123, 123, 123, 0.5);
  }

  .link > a:hover {
    text-decoration: none;
  }
  
  .ddm {
    color: steelblue;
    font-weight: bold;
    font-size: 1.2rem;
  }

  .brand-link {
    font-weight: bolder;
    font-size: 1.5rem;
  }

  .brand-name {
    color: Black;
    font-weight: bolder;
    font-size: 1.5rem;
  }
  .brand-name:hover {
    text-decoration: none;
  }