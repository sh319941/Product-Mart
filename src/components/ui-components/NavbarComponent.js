import React from 'react';
import { useHistory, Link } from "react-router-dom";
import { Navbar, Nav, NavDropdown } from "react-bootstrap";

import './NavbarComponent.css';

const NavbarComponent = () => {

  let history = useHistory();
  const renderAuthButton = () => {
    if (localStorage.getItem("username") === null) {
      return (
        <Navbar bg="light" expand="lg">
          <Navbar.Brand>
            <Link to="/" className="brand-name">
              Product Mart
            </Link>
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="ml-auto">
              <Nav.Link className="link">
                <Link to="/products">Products</Link>
              </Nav.Link>

              <Nav.Link className="link">
                <Link to="/addProduct">Add Product</Link>
              </Nav.Link>

              <Nav.Link className="link">
                <Link to="/login">Login</Link>
              </Nav.Link>
              <Nav.Link className="link">
                <Link to="/register">Register</Link>
              </Nav.Link>
            </Nav>
          </Navbar.Collapse>
        </Navbar>
      );
    } else {
      return (
        <Navbar bg="light" expand="lg">
          <Navbar.Brand>
            <Link to="/" className="brand-name">
              Product Mart
            </Link>
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="ml-auto">
              <Nav.Link className="link">
                <Link to="/products">Products</Link>
              </Nav.Link>  
              <Nav.Link className="link">
                <Link to="/myproducts">My Products</Link>
              </Nav.Link>
              <Nav.Link className="link">
                <Link to="/addProduct">Add Product</Link>
              </Nav.Link>
              <Nav.Link className="link">
                <Link to="/cart">Cart</Link>
              </Nav.Link>
              <Nav.Link className="link">
                <Link to="/productStats">Statistics</Link>
              </Nav.Link>
              <NavDropdown
                title={localStorage.getItem("username")}
                id="basic-nav-dropdown"
                className="ddm"
              >
                <NavDropdown.Item>
                  <Link to="/myprofile">My Profile</Link>
                </NavDropdown.Item>
                <NavDropdown.Divider />
                <NavDropdown.Item onClick={() => logout()}>
                  Logout
                </NavDropdown.Item>
              </NavDropdown>
            </Nav>
          </Navbar.Collapse>
        </Navbar>
      );
    }
  };

    function logout() {
      localStorage.clear();
      history.push("/");
    }

  return <div>{renderAuthButton()}</div>;
};

export default NavbarComponent;