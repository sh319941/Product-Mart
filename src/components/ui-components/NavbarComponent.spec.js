import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';

import NavbarComponent from './NavbarComponent';

describe('Testing Navbar Component', () => {

    it('Snapshot of Navbar component', () => {
        const wrapper = shallow(<NavbarComponent />);

        expect(wrapper).toMatchSnapshot();
    })

    it('check for links in navbar', () => {
        const wrapper = shallow(<NavbarComponent />);

        expect(wrapper.find(Link)).toHaveLength(5);
    })
})