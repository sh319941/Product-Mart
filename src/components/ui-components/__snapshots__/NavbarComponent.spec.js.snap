// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar Component Snapshot of Navbar component 1`] = `
<div>
  <Navbar
    bg="light"
    collapseOnSelect={false}
    expand="lg"
    variant="light"
  >
    <NavbarBrand>
      <Link
        className="brand-name"
        to="/"
      >
        Product Mart
      </Link>
    </NavbarBrand>
    <NavbarToggle
      aria-controls="basic-navbar-nav"
      label="Toggle navigation"
    />
    <NavbarCollapse
      id="basic-navbar-nav"
    >
      <Nav
        className="ml-auto"
        fill={false}
        justify={false}
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          className="link"
          disabled={false}
        >
          <Link
            to="/products"
          >
            Products
          </Link>
        </NavLink>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          className="link"
          disabled={false}
        >
          <Link
            to="/addProduct"
          >
            Add Product
          </Link>
        </NavLink>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          className="link"
          disabled={false}
        >
          <Link
            to="/login"
          >
            Login
          </Link>
        </NavLink>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          className="link"
          disabled={false}
        >
          <Link
            to="/register"
          >
            Register
          </Link>
        </NavLink>
      </Nav>
    </NavbarCollapse>
  </Navbar>
</div>
`;
